### ISSUE TITLE
a11y guidelines/issues on tooltip doc element

### ISSUE THREAD

**Issue by Aniort** [Orange-OpenSource/Orange-Boosted-Bootstrap#1636]
### Prerequisites

- [X] I have [searched](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

first of all, must be usable with a keyboard only:
- the triggering element must be a button (or role="button")
- this element must have an aria-descirbedby referencing the id of the tooltip only IF non interactive or semantic content 
- the tooltip container mut have a role="tooltip"
- When the focus (and hover) is on the tooltip triggering element aria-expanded goes from false to true to warn the visually disabled users via AT (assistive technologies) that something has happened on focus.

 and :
- The tooltip must be hidden when a keyboard user presses "Esc" (unless the tooltip will never overlap other content).
- Allow a mouse or pointer user to hide the tooltip, by activating the triggering element, if the tooltip content is not interactive or semantic (unless the tooltip will never overlap other content ), 
- The tooltip must allow a mouse user to move their mouse over the tooltip content without dismissing it.
- The tooltip remains in view until the user actively dismisses it (for example onblur if triggered by onfocus and mouseOut if mouseIn invoked the tooltip, so going out of the triggerng element), or it is no longer valid.

and also, if the content is interactive or semantic:
- add, as the first in:teractive element, a close button/disclosure (unless the tooltip will never overlap other content ) 
- Insert tabindex="0" on the tooltip container (to make it focusable)
- the tooltip element must be just after the triggering element in the code (next sibilling) or be sure that the next "tab" after the triggering element is the tooltip or popover element (and the inverse for Shift + "tab")

### Reduced test cases

i'll provide a proof of concept soon
the closer example must be: https://zoebijl.github.io/apg-tooltip/

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Chrome, Firefox, Microsoft Edge

### What version of Boosted are you using?

5.2.x

**Comment by MaxLardenois** [comment]
Work will be done in #3827 for OUDS
